src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -118,6 +117,15 @@ public void checkout(
             @CheckForNull SCMRevisionState baseline)
             throws IOException, InterruptedException {
         gitSCM.checkout(build, launcher, workspace, listener, changelogFile, baseline);
+
```

Comment:
Unfortunately, the listener cannot be used for this as `onInitialized` and `onStarted` call before the checkout is made. That said, I feel this functionality may be better encapsulated, perhaps (at least?) in a separate method that checkout calls.

Reply:
Currently `BitbucketSCM` is a wrapper on top of `GitSCM` and it would be better if we can keep it and avoid adding any extra code to it. What do you think about adding a `GitSCMExtension` which would post the build result in `GitSCMExtension.onCheckoutCompleted()` ?

Reply:
I have dug around a little, and.. we can do this slightly differently I think. 
I'm not took keen to have to wait for the checkout to complete before the 'in progress build status is posted. 
How about, we add a custom gitSCMExtesion that does a "decorateCheckoutCommand" that posts the build status. This runs *before* the checkout is invoked so if you have a long checkout (as some large projects do) the status will be updated before the checkout is complete. 
With a bit of software magic we don't need to expose our extension to the world. We can just inject it before creating the GitSCM and we have to filter it out before returning the list to the UI. 
Happy to talk to you about it

Reply:
> This runs before the checkout is invoked so if you have a long checkout (as some large projects do) the status will be updated before the checkout is complete.

I think you have mistaken `clone` with `checkout`. `checkout` should be pretty quick.
`decorateCheckoutCommand` doesn't look like being build for our use case, it surely can be used but it's an extension point to alter checkout command.